tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1932,6 +1970,76 @@ public:
    ~RField() override = default;
 };
 
+template <typename ItemT>
+class RField<std::set<ItemT>> : public RSetField {
+   using ContainerT = typename std::set<ItemT>;
+
+protected:
+   std::size_t AppendImpl(const Detail::RFieldValue &value) final
+   {
+      auto typedValue = value.Get<ContainerT>();
+      auto nbytes = 0;
+      auto count = typedValue->size();
+      for (auto item : *typedValue) {
+         auto itemValue = fSubFields[0]->CaptureValue(static_cast<void *>(&item));
+         nbytes += fSubFields[0]->Append(itemValue);
+      }
+      Detail::RColumnElement<ClusterSize_t, EColumnType::kUnknown> elemIndex(&this->fNWritten);
+      this->fNWritten += count;
+      fColumns[0]->Append(elemIndex);
+      return nbytes + sizeof(elemIndex);
+   }
+
+   void ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value) final
+   {
+      auto typedValue = value->Get<ContainerT>();
+      RClusterIndex collectionStart;
+      ClusterSize_t nItems;
+      fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nItems);
+      typedValue->clear();
+      for (unsigned i = 0; i < nItems; ++i) {
+         ItemT item;
+         auto itemValue = fSubFields[0]->CaptureValue(&item);
+         fSubFields[0]->Read(collectionStart + i, &itemValue);
+         typedValue->insert(item);
```

Comment:
I am a bit confused.   Why do we need to capture the item to send (indirectly?) pass it to `Read`.  (and/or why do we need a local object+CaptureValue rather than a`Get`)?

Reply:
Do you mean for set fields specifically, or for fields in general? If it's the latter, I mostly adapted the already existing code to work for sets (i.e. from `RVectorField` and `RRVecField`), so @jblomer or @jalopezg-git can probably give a better answer than I do.

Reply:
I meant for the content of collections.

Reply:
I guess we could just do it once before the loop in [line 2001](https://github.com/root-project/root/pull/12948/files#diff-4ee1577c07fac4b222a49c703265ff7b52e692dafa8f46b1ad4713ed526c82fcR2001), given that the underlying pointer doesn't change.